raiden/tests/utils/transfer.py

```diff
@@ -646,57 +646,48 @@ def assert_balance_proof(
         )
 
         if received_balance_proof is not None:
-            if type(received_balance_proof) == ReceiveTransferRefund:
```

Comment:
These were all wrong, it should be checking against `.data` and not `received_balance_proof` (since that is always a `StateChangeRecord`)